src/Compilers/VisualBasic/Portable/Compilation/VisualBasicCompilation.vb

```diff
@@ -2654,11 +2630,18 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                 i = i + 1
             Next
 
-            Dim fields = memberTypes.ZipAsArray(
-                memberNames,
-                Function(type, name) New AnonymousTypeField(name, DirectCast(type, TypeSymbol), Location.None))
+            Dim fields = ArrayBuilder(Of AnonymousTypeField).GetInstance()
+
+            For i = 0 To memberTypes.Length - 1
+                Dim type = memberTypes(i)
+                Dim name = memberNames(i)
+                Dim loc = If(memberLocations.IsDefault, Location.None, memberLocations(i))
+                Dim isReadOnly = memberIsReadOnly.IsDefault OrElse memberIsReadOnly(i)
```

Comment:
>memberIsReadOnly.IsDefault OrElse memberIsReadOnly(i) [](start = 33, length = 53)

I think this condition should be ```Not memberIsReadOnly.IsDefault AndAlso memberIsReadOnly(i)``` otherwise it looks like we are changing default behavior of this API.